seisbench/models/dkpn.py

```diff
@@ -0,0 +1,784 @@
+            sampling_rate=sampling_rate,
+            default_args={
+                "overlap": 1500,
+                "P_threshold": 0.2,
+                "S_threshold": 0.2,
+                #
+                "fp_stabilization": 4,  # sec
+                "t_long": 4,
+                "freqmin": 0.5,
+                "corner": 1,
+                "perc_taper": 0.1,
+                "mode": "rms",
+                "clip": -999,
+                "log": True,
+                "normalize": False,
+                "polarization_win_len": 1,
+                "use_amax_only": False},
+            **kwargs,
+        )
+
+        self.in_channels = in_channels
+        self.classes = classes
+        self.norm = norm
+        self.depth = 5
+        self.kernel_size = 7
+        self.stride = 4
+        self.filters_root = 8
+        self.activation = torch.relu
+
+        self.inc = nn.Conv1d(
+            self.in_channels, self.filters_root, self.kernel_size, padding="same"
+        )
+        self.in_bn = nn.BatchNorm1d(8, eps=1e-3)
+
+        self.down_branch = nn.ModuleList()
+        self.up_branch = nn.ModuleList()
+
+        last_filters = self.filters_root
+        for i in range(self.depth):
+            filters = int(2**i * self.filters_root)
+            conv_same = nn.Conv1d(
+                last_filters, filters, self.kernel_size, padding="same", bias=False
+            )
+            last_filters = filters
+            bn1 = nn.BatchNorm1d(filters, eps=1e-3)
+            if i == self.depth - 1:
+                conv_down = None
+                bn2 = None
+            else:
+                if i in [1, 2, 3]:
+                    padding = 0  # Pad manually
+                else:
+                    padding = self.kernel_size // 2
+                conv_down = nn.Conv1d(
+                    filters,
+                    filters,
+                    self.kernel_size,
+                    self.stride,
+                    padding=padding,
+                    bias=False,
+                )
+                bn2 = nn.BatchNorm1d(filters, eps=1e-3)
+
+            self.down_branch.append(nn.ModuleList([conv_same, bn1, conv_down, bn2]))
+
+        for i in range(self.depth - 1):
+            filters = int(2 ** (3 - i) * self.filters_root)
+            conv_up = nn.ConvTranspose1d(
+                last_filters, filters, self.kernel_size, self.stride, bias=False
+            )
+            last_filters = filters
+            bn1 = nn.BatchNorm1d(filters, eps=1e-3)
+            conv_same = nn.Conv1d(
+                2 * filters, filters, self.kernel_size, padding="same", bias=False
+            )
+            bn2 = nn.BatchNorm1d(filters, eps=1e-3)
+
+            self.up_branch.append(nn.ModuleList([conv_up, bn1, conv_same, bn2]))
+
+        self.out = nn.Conv1d(last_filters, self.classes, 1, padding="same")
+        self.softmax = torch.nn.Softmax(dim=1)
+
+        # DKPN attributes for taking care of windows CF and probs
```

Comment:
Any transient attributes should not be stored in class variables. From what I see, these variable are actually only written to but never read. Can they be deleted?